blockchain/contracts/HelloWorld.sol

```diff
@@ -1,52 +1,124 @@
 pragma solidity ^0.8.0;
 
+import "./AddrArrayLib.sol";
+
 contract HelloWorld {
+    using AddrArrayLib for AddrArrayLib.Addresses;
     address payable nftOwner;
     address public buyer;
 
+    uint256[] public availableNFTs;
 
-    uint256[] public nfts;
+    uint256 public nftsLength;
 
     struct NFTOwnership {
         address owner;
         uint256 nftId;
-        uint droptime;
+        uint256 dropTime;
+        uint256 dropHash;
+        address reservedFor;
     }
 
     mapping(uint256 => NFTOwnership) public nftOwnerships;
 
+    // Used to track which addresses have joined the drop
+    // address[] private joinedUsers;
+    AddrArrayLib.Addresses joinedUsers;
 
     constructor() {
         buyer == msg.sender;
         nftOwner = payable(0xC77787e364E3420c0609a249F18De47430900f0C);
     }
 
-    // create a NFT
-    function mockNFT(uint _droptime) public returns (uint256) {
-        uint256 nftHash = getNFTidentifier();
+    function createDrop(uint256 _dropTime) public {
+        uint256 dropHash = generateRandomNumber(0);
+        uint256 i;
+        // I choose "3" as a value for testing reasons
+        for (i = 0; i < 3; i++) {
+            mockNFT(_dropTime, dropHash, i + 1);
+        }
+    }
+
+    // create NFTs for drop
+    function mockNFT(
+        uint256 _dropTime,
+        uint256 _dropHash,
+        uint256 _number
+    ) public {
+        uint256 nftHash = generateRandomNumber(_number);
         NFTOwnership memory nftOwnership;
         nftOwnership.nftId = nftHash;
         nftOwnership.owner = nftOwner;
-        nftOwnership.droptime = _droptime;
+        nftOwnership.reservedFor = nftOwner;
+        nftOwnership.dropTime = _dropTime;
+        nftOwnership.dropHash = _dropHash;
         nftOwnerships[nftHash] = nftOwnership;
```

Comment:
when initializing, nftHash is used as Index

Reply:
What's the issue with that?